security/pkg/server/ca/server.go

```diff
@@ -117,9 +117,18 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 	return response, nil
 }
 
+func recordCertsExpiry(ca CertificateAuthority) {
+	caKeyCertBundler := ca.GetCAKeyCertBundle()
+	rootCertExpiryTimestamp.Record(extractRootCertExpiryTimestamp(caKeyCertBundler))
+
+	if len(caKeyCertBundler.GetCertChainPem()) > 0 {
+		certChainExpiryTimestamp.Record(extractCertChainExpiryTimestamp(caKeyCertBundler))
+	}
+}
+
 // extractRootCertExpiryTimestamp returns the unix timestamp when the root becomes expires.
-func extractRootCertExpiryTimestamp(ca CertificateAuthority) float64 {
-	rb := ca.GetCAKeyCertBundle().GetRootCertPem()
+func extractRootCertExpiryTimestamp(keyCertBundle util.KeyCertBundle) float64 {
```

Comment:
i feel  these two extraction function are closed enough to collapse into a single function, they always use together as code indicates now.

Reply:
I get you, but I believe that they have different responsibilities and the way that I found to group them was to create another function: `recordCertsExpiry`.

so I feel that although we are creating more functions, it's going to be easier to understand the code.

Reply:
Please move these functions to security/pkg/pki/util/keycertbundle.go so that they are the member functions of KeyCertBundle.